README.md

```diff
@@ -0,0 +1,195 @@
+thread you must call `xtm_queue_notify_consumer`. The less often you notify, the
+greater the performance, but the greater the latency.
+Returns 0 if queue has space. Otherwise -1 with errno set to `ENOBUFS`.
+
+## xtm_queue_consumer_fd
+
+Returns file descriptor, that should be watched by consumer thread to
+become readable. When it became readable, consumer should call one of
+the consumer functions: `xtm_queue_pop_ptrs` or `xtm_queue_invoke_funs_all`.
+
+## xtm_queue_producer_fd
+
+Return file descriptor, that should be watched by producer thread to become
+readable in case when `xtm_queue_push_fun` or `xtm_queue_push_ptr` fails. When
+it became readable, producer may try push to the queue again. Very rarely, there
+may be a situation (because of race between producer and consumer threads), when
+the descriptor became readable, but queue is still full. In this case producer
+thread, need to poll this descriptor again.
+
+## xtm_queue_invoke_funs_all
+
+Function calls all functions contained in the queue at the time this function is
+called. We do not call those functions, that the producer thread can add at the
+moment, when this function is already called, in order to prevent an infinite loop
+and hunger. Also, if the queue size is very large (16384 or more), then hunger is
+still possible, since this function does not allow limiting the number of called
+functions.
+Return count of invoked functions.
+
+## xtm_queue_push_ptr
+
+Function push pointer to the queue. This function does not notify the consumer
+thread, but only push to the queue. To notify the consumer thread you must call
+`xtm_queue_notify_consumer`. The less often you notify, the greater the performance,
+but the greater the latency.
+Returns 0 if queue has space. Otherwise -1 with errno set to `ENOBUFS`.
+
+## xtm_queue_pop_ptrs
+
+Function gets up to count elements from the queue and save them in pointer array.
+Return count of extracted messages.
+
+## xtm_queue_consume
+
+Function flush queue pipe, according to file descriptor passed to this function.
+Return 0 on success. Otherwise -1 with errno set appropriately (as in `read(2)`,
+since it implies read from internal fd).
+
+## xtm_queue_was_full
+
+Function check hat queue was full, and producer thread fails to put to the queue.
+Return true if queue was full, otherwise return false
+
+Examples
+--------
+
+Full example you can see in test or perf repositories, here is only simple
+example of using library.
+
+** Allocate xtm queue with appropriate flags **
+
+```c
+unsigned flags = 0;
+flags |= XTM_QUEUE_NEED_TO_CLOSE_READFD;
+flags |= XTM_QUEUE_PRODUCER_NEED_NOTIFICATIONS;
+xtm_queue = xtm_queue_new(XTM_QUEUE_SIZE, flags);
+```
+
+** Get internal fd on consumer side and pool it in the loop **
+
+```c
+int fd = xtm_queue_consumer_fd(xtm_queue);
+fd_set readset;
+FD_ZERO(&readset);
+FD_SET(fd, &readset);
+select(fd + 1, &readset, NULL, NULL, NULL);
```

Comment:
Please remove select() from readme/examples. It is absolutely unusable in large-scale systems (file descriptors values exceed 1023 quickly).

Reply:
Change to poll, fixed